tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/annotations/annotation.bal

```diff
@@ -27,6 +27,48 @@ function testAnnotOnBoundMethod() {
     assertEquality(102, rec.i);
 }
 
+type AnnotationRecord record {|
+    int minValue;
+    int maxValue;
+|};
+
+annotation AnnotationRecord BarAnnotation on type ;
+annotation AnnotationRecord BarAnnotation2 on type ;
+
+@BarAnnotation {
+    minValue: 18,
+    maxValue: 36
+}
+type Bar int|float;
+
+function testAnnotationWithReferredType() {
+    typedesc<any> 'type = typeof Bar;
```

Comment:
`typedesc td = Bar;` results in the typedesc of `Bar` (e.g., `typedesc<record {}>`).

In `typedesc td = typeof Bar;`, `typeof Bar` results in a `typedesc` of a `typedesc` (e.g., `typedesc<typedesc<record {}>>`). I'm not sure if the current behaviour is correct.

Reply:
I updated the tests according to the https://github.com/ballerina-platform/ballerina-lang/pull/38120#discussion_r1006464228.